currency-rate/internal/server/service/rate.go

```diff
@@ -0,0 +1,40 @@
+package service
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/currency-rate/internal/rate"
+)
+
+type RateFetcher interface {
+	FetchRate(ctx context.Context, ccFrom, ccTo string) (rate.Rate, error)
+}
+
+type RateService struct {
+	fetchers []RateFetcher
+}
+
+var _ RateFetcher = (*RateService)(nil) // Ensure RateService implements RateFetcher
```

Comment:
what this line of code is doing?

Reply:
This checks that RateService also complies with the RateFetcher interface

Reply:
https://go.dev/doc/effective_go#blank_implements